src/ILVerification/src/InterfaceVerifier.cs

```diff
@@ -0,0 +1,157 @@
+                foreach (InterfaceImplementationHandle interfaceHandle in interfaceHandles)
+                {
+                    InterfaceImplementation interfaceImplementation = _module.MetadataReader.GetInterfaceImplementation(interfaceHandle);
+                    DefType interfaceType = _module.GetType(interfaceImplementation.Interface) as DefType;
+                    if (interfaceType == null)
+                    {
+                        ThrowHelper.ThrowTypeLoadException(ExceptionStringID.ClassLoadBadFormat, type);
+                    }
+
+                    InterfaceMetadataObjects imo = new InterfaceMetadataObjects
+                    {
+                        DefType = interfaceType,
+                        InterfaceImplementationHandle = interfaceHandle
+                    };
+
+                    if (!implementedInterfaces.Contains(imo))
+                    {
+                        implementedInterfaces.Add(imo);
+                    }
+                    else
+                    {
+                        InterfaceVerificationResult?.Invoke(new InterfaceVerificationResult()
+                        {
+                            Type = _typeDefinitionHandle,
+                            Error = new InterfaceVerificationErrorArgs()
+                            {
+                                Code = VerifierError.InterfaceImplHasDuplicate,
+                                TokenClass = _module.MetadataReader.GetToken(_typeDefinitionHandle),
+                                TokenInterface = _module.MetadataReader.GetToken(interfaceHandle)
+                            },
+                            Message = string.Format(_stringResourceManager.Value.GetString(VerifierError.InterfaceImplHasDuplicate.ToString(), CultureInfo.InvariantCulture), interfaceType.ToString())
+                        });
+                    }
+                }
+
+                // Looks for missing method implementation
+                foreach (InterfaceMetadataObjects interfaceMetadataObjects in implementedInterfaces.Distinct())
+                {
+                    foreach (MethodDesc method in interfaceMetadataObjects.DefType.GetAllMethods())
+                    {
+                        if (!IsImplemented(type, method))
+                        {
+                            InterfaceVerificationResult?.Invoke(new InterfaceVerificationResult()
+                            {
+                                Type = _typeDefinitionHandle,
+                                Error = new InterfaceVerificationErrorArgs()
+                                {
+                                    Code = VerifierError.InterfaceMethodNotImplemented,
+                                    TokenClass = _module.MetadataReader.GetToken(_typeDefinitionHandle),
+                                    TokenInterface = _module.MetadataReader.GetToken(interfaceMetadataObjects.InterfaceImplementationHandle)
```

Comment:
I want to include Method token on ui log in case of fail(like PEVerify)...but I cannot find a way to get token(int) from MethodDesc

Reply:
/cc @jkotas @MichalStrehovsky @jcouv

Reply:
>  get token(int) from MethodDesc

You need to cast the MethodDesc to `EcmaMethod` and then get the token from `MethodDefinitionHandle`.